packages/astro/src/core/render-context.ts

```diff
@@ -93,18 +93,22 @@ export class RenderContext {
 	 */
 	async render(
 		componentInstance: ComponentInstance | undefined,
-		slots: Record<string, any> = {}
+		slots: Record<string, any> = {},
+		props: Record<string, any> = {}
 	): Promise<Response> {
 		const { cookies, middleware, pathname, pipeline } = this;
 		const { logger, routeCache, serverLike, streaming } = pipeline;
-		const props = await getProps({
-			mod: componentInstance,
-			routeData: this.routeData,
-			routeCache,
-			pathname,
-			logger,
-			serverLike,
-		});
+		Object.assign(
+			props,
+			await getProps({
+				mod: componentInstance,
+				routeData: this.routeData,
+				routeCache,
+				pathname,
+				logger,
+				serverLike,
+			})
+		);
```

Comment:
`props` are now part of the class `RenderContext`, so we should update this code